### PR TITLE
Add formula for OSRM

### PR DESCRIPTION
**Pull request by tmcw**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/osrm-backend.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  def install
+    mkdir "build" do
+      system "cmake", "..", "-DCMAKE_BUILD_TYPE=Release", *std_cmake_args
+      system "cmake", "--build", "."
+      system "cmake", "--build", ".", "--target", "install"
+    end
+  end
+
+  test do
+    system "npm install"
```
Comment (mistydemeo) [inline Formula/osrm-backend.rb]: This test block doesn't run within the project source, so there's nothing to `npm install`.

File: Formula/osrm-backend.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+class OsrmBackend < Formula
+  desc "High performance routing engine"
+  homepage "http://project-osrm.org/"
+  url "https://github.com/Project-OSRM/osrm-backend/archive/v5.4.0.tar.gz"
+  sha256 "6393745fe8250ad0387b1fe3f4a2218cf692528acb55a4e0198ba5fbe1c81231"
+
+  depends_on "cmake" => :build
+  depends_on "node" => :build
+  depends_on "boost"
```
Comment (mistydemeo) [inline Formula/osrm-backend.rb]: Do you mind alphabetizing the non-build dependencies?

File: Formula/osrm-backend.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  depends_on "cmake"
+  depends_on "libzip"
+  depends_on "libstxxl"
+  depends_on "libxml2"
+  depends_on "lua51"
+  depends_on "luabind"
+  depends_on "tbb"
+  depends_on "ccache"
```
Comment (mistydemeo) [inline Formula/osrm-backend.rb]: Is ccache required, or just here to speed up the build?

**Review by mistydemeo — CHANGES_REQUESTED** [review]

File: Formula/osrm-backend.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  def install
+    mkdir "build" do
+      system "cmake", "..", "-DCMAKE_BUILD_TYPE=Release", *std_cmake_args
```
Comment (MikeMcQuaid) [inline Formula/osrm-backend.rb]: @daniel-j-h We should rely on it because it won't change at any time.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/osrm-backend.rb (hunk 3 above)
Comment (tmcw) [inline Formula/osrm-backend.rb]: 👍 ccache is removed

File: Formula/osrm-backend.rb (hunk 2 above)
Comment (tmcw) [inline Formula/osrm-backend.rb]: 👍  dependencies sorted

File: Formula/osrm-backend.rb (hunk 1 above)
Comment (tmcw) [inline Formula/osrm-backend.rb]: 👍 replaced this with standalone tests that don't require node and use the built program

File: Formula/osrm-backend.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      system "cmake", "--build", "."
+      system "cmake", "--build", ".", "--target", "install"
```
Comment (tmcw) [inline Formula/osrm-backend.rb]: 👍 fixed

File: Formula/osrm-backend.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    end
+  end
+
+  test do
+    system "npm install"
+    system "npm test"
```
Comment (tmcw) [inline Formula/osrm-backend.rb]: 👍 replaced with a functional test

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**